ui/src/criteria/classification.tsx

```diff
@@ -709,12 +767,49 @@ async function search(
   return results.flat();
 }
 
-function firstOf(value: string | string[]) {
-  // TODO(tjennison): Multiple occurrence: This uses the first occurrence
-  // instead of somehow taking them all account, assuming that they're
-  // interchangeable.
-  if (typeof value === "string") {
-    return value;
+function useEntityData(
+  config: Config
+): [boolean, EntityGroupData[], EntityGroupData[]] {
+  const underlaySource = useUnderlaySource();
+
+  return useMemo(() => {
+    const classificationEntityGroupData = (
+      config.classificationEntityGroups ?? []
+    ).map((eg) => underlaySource.lookupEntityGroup(eg.id));
+
+    const groupingEntityGroupData = (config.groupingEntityGroups ?? []).map(
+      (eg) => underlaySource.lookupEntityGroup(eg.id)
+    );
+
+    const hasHierarchies = [
+      ...classificationEntityGroupData,
+      ...groupingEntityGroupData,
+    ].reduce((h, eg) => h || !!eg.selectionEntity.hierarchies?.length, false);
+
+    return [
+      hasHierarchies,
+      classificationEntityGroupData,
+      groupingEntityGroupData,
+    ];
+  }, [config.classificationEntityGroups, config.groupingEntityGroups]);
+}
+
+function lookupEntityGroupData(list: EntityGroupData[], id: string) {
+  return list.find((eg) => eg.id === id);
+}
+
+function configEntityGroups(config: Config) {
+  const entityGroups = [
+    ...(config.classificationEntityGroups ?? []),
+    ...(config.groupingEntityGroups ?? []),
+  ];
+  if (!entityGroups) {
+    throw new Error("At least one entity group must be configured.");
   }
-  return value[0];
+  return entityGroups;
 }
+
+const DEFAULT_SORT_ORDER = {
+  attribute: ROLLUP_COUNT_ATTRIBUTE,
+  direction: SortDirection.Desc,
+};
```

Comment:
Not for this PR, but it would be nice to have these defaults defined in the config schema so they show up in documentation. Ditto for the `DEFAULT_LIMIT` above.

Reply:
As long as they would be guaranteed to be defined there. These exist as a fallback for when no value is specified.